### PR TITLE
Ledger API tests: option for non-unique identifiers

### PR DESCRIPTION
**Pull request by gleber-da**
Added new CLI flags `--stable-party-identifiers` and `--stable-command-identifiers` to the Ledger API Test Tool to allow disabling randomization of party and command identifiers. It is useful for testing of ledgers which are configured with a predefined static set of parties. This is supported by appropriate `PlatformApplications.Config` fields and plumbing in `TransactionServiceIT` and `SandboxSemanticTestsLfRunner`.

### Pull Request Checklist

- [x] Read and understand the [contribution guidelines](https://github.com/digital-asset/daml/blob/master/CONTRIBUTING.md)
- [x] Include appropriate tests
- [x] Set a descriptive title and thorough description
- [x] Add a reference to the [issue this PR will solve](https://github.com/digital-asset/daml/issues), if appropriate
- [x] Add a line to the [release notes](https://github.com/digital-asset/daml/blob/master/docs/source/support/release-notes.rst), if appropriate

NOTE: CI is not automatically run on non-members pull-requests for security
reasons. The reviewer will have to comment with `/AzurePipelines run` to
trigger the build.

### CODE REVIEW

File: ledger/ledger-api-integration-tests/src/test/itsuite/scala/com/digitalasset/platform/tests/integration/ledger/api/TransactionServiceIT.scala [hunk 1]
```diff
@@ -324,7 +319,7 @@ class TransactionServiceIT
       "expose transactions to non-submitting stakeholders without the commandId" in allFixtures {
         c =>
           c.submitCreateAndReturnTransaction(
-              s"Checking_commandId_visibility_for_non-submitter_party-${runSuffix}",
+              s"Checking_commandId_visibility_for_non-submitter_party-${runCommandSuffix}",
```
Comment (gerolf-da) [inline ledger/ledger-api-integration-tests/src/test/itsuite/scala/com/digitalasset/platform/tests/integration/ledger/api/TransactionServiceIT.scala]: The leading `-` needs to be removed now from all command IDs.

File: ledger/test-common/src/main/scala/com.digitalasset.platform.testing/MultiFixtureBase.scala [hunk 2]
```diff
@@ -38,7 +39,7 @@ trait MultiFixtureBase[FixtureId, TestContext]
     def context(): TestContext = createContext()
   }
 
-  def timeLimit: Span = 30.seconds
+  def timeLimit: Span = 60.seconds * spanScaleFactor
```
Comment (gerolf-da) [inline ledger/test-common/src/main/scala/com.digitalasset.platform.testing/MultiFixtureBase.scala]: Why the increased timeout?

**Comment by gleber-da** [comment]
@gaboraranyossy-da @gerolf-da PTAL

File: ledger/ledger-api-integration-tests/src/test/itsuite/scala/com/digitalasset/platform/tests/integration/ledger/api/DivulgenceIT.scala [hunk 3]
```diff
@@ -172,7 +174,7 @@ class DivulgenceIT
   private def createDivulgence2(ctx: LedgerContext, workflowId: String): Future[ContractIdString] =
     create(
       ctx,
-      s"create-Divulgence2-$runSuffix",
+      s"create-Divulgence2-${runCommandSuffix}",
```
Comment (gerolf-da) [inline ledger/ledger-api-integration-tests/src/test/itsuite/scala/com/digitalasset/platform/tests/integration/ledger/api/DivulgenceIT.scala]: Use `commandIdUniifier` here?

File: ledger/ledger-api-integration-tests/src/test/lib/scala/com/digitalasset/platform/apitesting/TestIdsGenerator.scala [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+// SPDX-License-Identifier: Apache-2.0
+
+package com.digitalasset.platform.apitesting
+
+import scala.util.Random
+
+/**
+  * Provides a mechanism to randomize various identifiers used in testing per run of a process.
+  *
+  * It enables Ledger API Test Tool to be ran against a persistent Ledger API endpoint repeatedly by avoiding
```
Comment (gerolf-da) [inline ledger/ledger-api-integration-tests/src/test/lib/scala/com/digitalasset/platform/apitesting/TestIdsGenerator.scala]:
```suggestion
  * It enables Ledger API Test Tool to be run against a persistent Ledger API endpoint repeatedly by avoiding
```

File: ledger/ledger-api-integration-tests/src/test/lib/scala/com/digitalasset/platform/apitesting/TestIdsGenerator.scala [hunk 5]
```diff
@@ -0,0 +1,26 @@
...
+trait TestIdsGenerator {
+  self: MultiLedgerFixture =>
+
+  lazy val runSuffix = Random.alphanumeric.take(10).mkString
+  lazy val runCommandSuffix = if (config.uniqueCommandIdentifiers) "-" + runSuffix else ""
+  // Using runCommandSuffix, since there does not seem to be a need to treat workflow ids differently than command ids.
+  // Yet lets wrap those identifiers into an appropriate function so we could do it in the future.
+  lazy val runWorkflowSuffix = runCommandSuffix
+  lazy val runPartySuffix = if (config.uniquePartyIdentifiers) "-" + runSuffix else ""
+  def partyNameUnifier(partyText: String) = partyText + runPartySuffix
```
Comment (gerolf-da) [inline ledger/ledger-api-integration-tests/src/test/lib/scala/com/digitalasset/platform/apitesting/TestIdsGenerator.scala]: It's not clear what this function "unifies". Maybe drop the `Unifier` suffix or rename it to `testPartyName`, etc...?

Comment (gleber-da) [reply ledger/ledger-api-integration-tests/src/test/lib/scala/com/digitalasset/platform/apitesting/TestIdsGenerator.scala]: Yes, I was also not happy with the name. `testPartyName` sounds so much better!

File: ledger/ledger-api-integration-tests/src/test/itsuite/scala/com/digitalasset/platform/tests/integration/ledger/api/DivulgenceIT.scala (hunk 3 above)
Comment (gleber-da) [reply ledger/ledger-api-integration-tests/src/test/itsuite/scala/com/digitalasset/platform/tests/integration/ledger/api/DivulgenceIT.scala]: Done

File: ledger/ledger-api-integration-tests/src/test/lib/scala/com/digitalasset/platform/apitesting/TestIdsGenerator.scala (hunk 4 above)
Comment (gleber-da) [reply ledger/ledger-api-integration-tests/src/test/lib/scala/com/digitalasset/platform/apitesting/TestIdsGenerator.scala]: Done

File: ledger/ledger-api-integration-tests/src/test/itsuite/scala/com/digitalasset/platform/tests/integration/ledger/api/TransactionServiceIT.scala (hunk 1 above)
Comment (gleber-da) [reply ledger/ledger-api-integration-tests/src/test/itsuite/scala/com/digitalasset/platform/tests/integration/ledger/api/TransactionServiceIT.scala]: Done

**Comment by gleber-da** [comment]
@gerolf-da All comments addressed, including the offline one from @gaboraranyossy-da

**Comment by gleber-da** [comment]
@gerolf-da Done. PTAL

**Pull request merged**